### DISCUSSION TITLE
Changing style of prompt on failure (not just $character)?

### DISCUSSION THREAD

**Discussion by cogikyo** [starship/starship#1998] [General]
I'm curious if there is a way to change style (color, mainly) of the prompt on a failed command. I looked through discussions and issues and didn't see it mentioned anywhere, as far as I can tell.

**Comment by mickimnet** [comment]
Hello @cullyn-inverba, the default prompt should turn red on a failure. See [configuration->character](https://starship.rs/config/#character). What happens, if enter `false` in your shell?

**Reply by mickimnet** [reply]
Or, where you referring to the whole prompt? That is not possible.

**Reply by cogikyo** [reply]
Yes, the whole promt, or at least selective parts of it, as character module can not be multi line and cannot contain other modules inside of it.

Thanks for answer anyways. Figured it wasn't currently possible but wanted to check. If I were to go about trying to contribute this functionality, or just do a mediocre hack, then where might I look?

**Reply by mickimnet** [reply]
On the top of my head, I cannot think of a hack. That would need a new module, IMHO.

**Reply by chipbuster** [reply]
Unfortunately, starship is designed as a set of relatively independent modules by design--we have very few truly global prompt settings by design.

Double unfortunately, modules do not return structured information by default--they return an ANSI string instead.

If you wanted to do a hack, look in `print.rs`. There, we call a function called `ansi_strings_for_shell` which essentially looks in the prompt string for anything that looks like an ANSI color escape code and wraps it in the shell-appropriate zero-length escape sequences. You could write a similar function that looks for anything like an ANSI color escape code and transforms them all into escapes for red color. Unfortunately, such a solution would have a pretty low chance of getting merged, since the `ansi_strings_for_shell` is already a kludge, and I'd be unwilling to add another such piece of code into the codebase just for turning everything red.

In the hypothetical starship refactor which will come out as 2.0 (and we're not even 1.0 yet, so this is a ways off!), the modules will return structured data which will be rendered into text by some frontend component. This should make it much easier to implement something like this in a sane manner, as you could just loop through the structured data and change the output color to red.

**Reply by cogikyo** [reply]
Excellent, thanks for the pointers! I looked a bit but had no luck getting a similar function to work. It's a super minor point, the character module is more than enough.
